aten/src/ATen/functorch/BatchRulesBinaryOps.cpp

```diff
@@ -287,6 +287,29 @@ void fill__Tensor_batch_rule(
   std::get<0>(self_and_other).copy_(std::get<1>(self_and_other));
 }
 
+std::tuple<Tensor, optional<int64_t>> log_sigmoid_backward_batch_rule(
+  Tensor& grad, optional<int64_t> grad_bdim,
+  Tensor& self, optional<int64_t> self_bdim,
+  Tensor& buffer, optional<int64_t> buffer_bdim) {
+  // NB: This emulates handle_pointwise_ops except we ignore the last argument, buffer
+  // when any of the inputs are on cuda.
+  // We do this because on cuda, buffer is a dummy tensor always of logical rank 1 and
+  // it becomes an issue when the rest of the inputs are scalar
+  int64_t out_logical_rank = std::max(rankWithoutBatchDim(grad, grad_bdim), rankWithoutBatchDim(self, self_bdim));
+  if (!grad.is_cuda() && !self.is_cuda() && !buffer.is_cuda()) {
+    out_logical_rank = std::max(out_logical_rank, rankWithoutBatchDim(buffer, buffer_bdim));
+  }
+  Tensor out_grad = maybePadToLogicalRank(moveBatchDimToFront(grad, grad_bdim), grad_bdim, out_logical_rank);
+  Tensor out_self =  maybePadToLogicalRank(moveBatchDimToFront(self, self_bdim), self_bdim, out_logical_rank);
+  Tensor out_buffer =  maybePadToLogicalRank(moveBatchDimToFront(buffer, buffer_bdim), buffer_bdim, out_logical_rank);
```

Comment:
nit: there's an extra space between the = and the m. Not sure why lintrunner doesn't catch this (do we not maintain style for ATen code?)